flyteidl/protos/flyteidl/core/workflow.proto

```diff
@@ -118,7 +118,7 @@ message ArrayNode {
     // point. Note that this is an optimistic restriction and that, due to network partitioning or
     // other failures, the actual number of currently running instances might be more. This has to
     // be a positive number if assigned. Default value is size.
-    uint32 parallelism = 2;
+    int64 parallelism = 2;
```

Comment:
We need to validate that this doesn't break registration as the IDL representation of a workflow will change. I'm not sure if uint32 and int64 will hash to the same value in flyteadmin. Basically to do this, we (1) write a workflow with an ArrayNode (with parallelism set) and execute using the existing Flyte then (2) update to include these changes and run the same workflow without changing anything. If everything runs smoothly, then lets merge.

Reply:
uint32 should hash to the same value in int64. Will confirm with running workflows and then merge if there aren't any issues.

Reply:
Master:
default:
{"json":{"exec_id":"f4917f6f67bfe4c01b18","node":"n0","ns":"flytesnacks-development","res_ver":"81806","routine":"worker-3","src":"handler.go:256","wf":"flytesnacks:development:tests.flytekit.integration.map_task.map_workflow"},"level":"info","msg":"ArrayNode with id [n0] has [0] parallelism","ts":"2024-04-04T12:33:49-07:00"}


Set concurrency 1:
{"json":{"exec_id":"fe481d782c28c4e9b9b0","node":"n0","ns":"flytesnacks-development","res_ver":"81866","routine":"worker-1","src":"handler.go:256","wf":"flytesnacks:development:tests.flytekit.integration.map_task.map_workflow"},"level":"info","msg":"ArrayNode with id [n0] has [1] parallelism","ts":"2024-04-04T12:34:45-07:00"}


Set concurrency -1:
Error: Value out of range: -1

Branch:
Default:
{"json":{"exec_id":"f8a424ddbe1754e96b45","node":"n0","ns":"flytesnacks-development","res_ver":"82148","routine":"worker-0","src":"handler.go:260","wf":"flytesnacks:development:tests.flytekit.integration.map_task.map_workflow"},"level":"info","msg":"ArrayNode with id [n0] has [0] parallelism","ts":"2024-04-04T12:39:53-07:00"}

Set concurrency 1:
{"json":{"exec_id":"f7f2fbfcc2ee04fd4860","node":"n0","ns":"flytesnacks-development","res_ver":"82203","routine":"worker-8","src":"handler.go:260","wf":"flytesnacks:development:tests.flytekit.integration.map_task.map_workflow"},"level":"info","msg":"ArrayNode with id [n0] has [1] parallelism","ts":"2024-04-04T12:40:24-07:00"}


Set concurrency -1:
{"json":{"exec_id":"f32c792fd0f9148bd9b3","node":"n0","ns":"flytesnacks-development","res_ver":"82348","routine":"worker-0","src":"handler.go:260","wf":"flytesnacks:development:tests.flytekit.integration.map_task.map_workflow"},"level":"info","msg":"ArrayNode with id [n0] has [-1] parallelism","ts":"2024-04-04T12:43:13-07:00"}

Reply:
Perfect, thanks for verifying!